settings: enable net/trace initially

tbg opened the pull request:
Override the net/trace setting to true so that we get traces at least until we
get the authoritative setting from the settings table. This is useful to allow
diagnosing a server that doesn't manage to connect to the rest of the cluster,
and in particular, we can always see the "server start" trace that tracks the
node's boot process, which can be useful to diagnose slow boot times.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/17944)
<!-- Reviewable:end -->

petermattis commented:
The rationale for doing this seems good to me. I'm not entirely sure if `Override` is the right mechanism to be using. @dt?

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17944#-:-KsZx3RKhTK7tF1nwXuZ:b-5dfj8b)*
<!-- Sent from Reviewable.io -->

a-robinson reviewed (approved):
I completely agree with the change in behavior. I'm not a huge fan of using a method marked as test-only in real code, though. Maybe you should at least update the comments on the `Override` methods to explain when they're safe to use.

dt commented:
Hm. The pattern used for this in the past has been to change the statically defined default, then write the intended default to the table via a migration -- since your default is used until you read the written value. If we want to use `Override` instead, we should ensure we have coverage of these semantics (swapped back on first read) in our tests and doc it.

tschottdorf commented:
Thanks for reminding me that this is the same kind of thing as the reporting options. PTAL!

dt reviewed (approved):
The downside of the explicitly wrt

dt commented:
Ugh, stray click on submit.

Downside to the explicitly written value approach we used before is that if we later change our mind about the default, we can't flip it in code for users who ran this migration... if we think that matters, then maybe your first approach makes sense. The reporting setting has the skip option so it has an extra argument in favor of writing (and conditionally *not* writing) an explicit value.

I don't feel strongly either way, just that if we use the in-mem override we a) doc and test those semantics b) doc when to use that pattern vs the migration swapped explicitly written value pattern.

Again though, LGTM either way.

tbg committed:
settings: enable net/trace initially

Default the net/trace setting to true with a sql migration that flips it back,
so that we get traces at least until we get the authoritative setting from the
settings table. This is useful to allow diagnosing a server that doesn't manage
to connect to the rest of the cluster, and in particular, we can always see the
"server start" trace that tracks the node's boot process, which can be useful to
diagnose slow boot times.

tschottdorf commented:
Yeah, doesn't matter in this case. TFTR!
